src/Moq/Mock.cs

```diff
@@ -598,29 +566,59 @@ private static Expression GetValueExpression(object value, Type type)
 			return Expression.Convert(Expression.Constant(value), type);
 		}
 
-		internal static SequenceSetup SetupSequence(Mock mock, LambdaExpression expression)
+		[DebuggerStepThrough]
+		private static TSetup SetupRecursive<TSetup>(Mock mock, LambdaExpression expression, Func<LambdaExpressionPart, Mock, TSetup> setupLast)
 		{
-			if (expression.IsProperty())
+			Debug.Assert(mock != null);
+			Debug.Assert(expression != null);
+			Debug.Assert(setupLast != null);
+
+			var parts = expression.Split();
+			return PexProtector.Invoke(SetupRecursivePexProtected, mock, expression, parts, setupLast);
```

Comment:
It's not so clear that we still need `PexProtector` in this place. After all, there's no longer any expression compilation going on, and we're not protecting expression tree operations elsewhere. As it's really difficult to know where to use `PexProtector` without having Pex-based tooling available (i.e. Visual Studio Ultimate with its IntelliTest feature), it may be best to remove `PexProtector` altogether once we've converted `SetupSet` and `VerifySet` to the new recursive / expression-rewriting approach.